ibis/backends/sql/compiler.py

```diff
@@ -1512,6 +1512,33 @@ def visit_Unsupported(self, op, **_):
             f"{type(op).__name__!r} operation is not supported in the {self.dialect} backend"
         )
 
+    def visit_TableUnnest(self, op, *, parent, column, offset: str | None):
+        quoted = self.quoted
+        parent_schema = op.parent.schema
+        parent_alias = parent.alias_or_name
+        column_name = op.column.name
+
+        column_alias = sg.to_identifier(column_name, quoted=quoted)
+        columns = [
+            sg.column(k, parent_alias, quoted=quoted)
+            if k != column_name
+            else column_alias
+            for k in parent_schema.keys()
+        ]
+        if column_name not in parent_schema:
+            columns.append(column_alias)
+
+        if offset is not None:
+            offset = sg.to_identifier(offset, quoted=quoted)
+            columns.append(offset)
+
+        expr = sg.select(*columns)
+        alias = sge.TableAlias(columns=[column_alias])
+        return expr.from_(parent).join(
+            sge.Unnest(expressions=[column], alias=alias, offset=offset),
+            how="CROSS",
```

Comment:
Any chance we could add a `how` parameter? It'd be useful to have `LEFT` join support so BigQuery DataFrames can better emulate pandas's explode method.

Reply:
Or a boolean parameter `preserve_empty_arrays` or something? If true, then it does a left join.

Reply:
added a `keep_empty` parameter, defaulting to `False`